### PR TITLE
chore: fix deprecation warnings and re-enable tracing with 0 sampling

### PR DESCRIPTION
**Pull request by omBratteng**
I have verified on local environment if I set the traceparent header to force sampling, it will let us trigger tracing when sampling ratio is 0.

### CODE REVIEW

**Review by rebelchris — APPROVED** [review]

File: src/telemetry/common.ts [hunk 1]
```diff
@@ -1,19 +1,16 @@
 import { FastifyRequest } from 'fastify';
-import dc from 'node:diagnostics_channel';
-import { SemanticAttributes as SemAttr } from '@opentelemetry/semantic-conventions';
+export { default as dc } from 'node:diagnostics_channel';
+export * from '@opentelemetry/semantic-conventions';
```
Comment (capJavert) [inline src/telemetry/common.ts:3]: Why are we re-exporting 3party packages?

Comment (capJavert) [reply src/telemetry/common.ts:3]: If we were to do it I would rather make it an abstraction layer/class/function that hides implementation and allows easy change without refactor. If its just due shorter lines I would rather keep it as is.

Comment (capJavert) [reply src/telemetry/common.ts:3]: If we were to do it I would rather make it an abstraction layer/class/function that hides implementation and allows easy change without refactor. Just changing imports I would rather leave it as is.

Comment (omBratteng) [reply src/telemetry/common.ts:3]: I'll remove the re-export of these two

Comment (omBratteng) [reply src/telemetry/common.ts:3]: :triggered: it doesn't want to give me a suggestion for auto-import of them
image

Comment (omBratteng) [reply src/telemetry/common.ts:3]: Needed to be a direct dependency for auto suggestions to work 😬

**Review by capJavert — APPROVED** [review]

**Pull request merged**